upload.go

```diff
@@ -1,40 +1,57 @@
 package s3
 
-import (
-	"context"
-	"fmt"
-	"io"
+import "io"
+
+// Upload represents a file that can be uploaded to the s3.
+type Upload interface {
+	io.ReadSeeker
+	Path() string
+	ContentType() string
+	MetaData() map[string]string
+	Size() *int64
+}
```

Comment:
Muss das überhaupt ein Interface sein? Wieso können wir nicht einfach nen Struct in die Upload function usw geben? 😄

Reply:
muss nicht, aber villeicht will man da selbst noch was implementieren 🤷🏻‍♂️ kann ich mir gut vorstellen

Reply:
kann ich mir irgendwie gar nicht vorstellen 😄

Reply:
naja ist ja jetzt nicht mehr so